lib/stackConfiguration/src/test/java/org/sagebionetworks/SynapseGoogleCloudStorageClientImplUnitTest.java

```diff
@@ -0,0 +1,180 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.sagebionetworks.gcp.SynapseGoogleCloudStorageClientImpl;
+
+import com.google.cloud.RestorableState;
+import com.google.cloud.WriteChannel;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.HttpMethod;
+import com.google.cloud.storage.Storage;
+
+@ExtendWith(MockitoExtension.class)
+public class SynapseGoogleCloudStorageClientImplUnitTest {
+	
+	@Mock
+	private Storage mockStorage;
+
+	@Mock
+	private Blob mockBlob;
+
+	@Captor
+	private ArgumentCaptor<Storage.SignUrlOption> signUrlMethodOptionCaptor;
+
+	@Captor
+	private ArgumentCaptor<Storage.SignUrlOption> signUrlTypeOptionCaptor;
+
+	@Captor
+	private ArgumentCaptor<Storage.ComposeRequest> composeRequestCaptor;
+
+
+	private SynapseGoogleCloudStorageClientImpl client;
+
+	private static final String BUCKET_NAME = "bucket-name";
+	private static final String OBJECT_KEY = "gcp-object-key";
+	private static final BlobId OBJECT_BLOB_ID = BlobId.of(BUCKET_NAME, OBJECT_KEY);
+	private static final BlobInfo OBJECT_BLOB_INFO = BlobInfo.newBuilder(OBJECT_BLOB_ID).build();
+
+	@BeforeEach
+	public void before() {
+		client = new SynapseGoogleCloudStorageClientImpl(mockStorage);
+	}
+	
+	@Test
+	public void getObject() {
+		when(mockStorage.get(OBJECT_BLOB_ID)).thenReturn(mockBlob);
+		Blob result = client.getObject(BUCKET_NAME, OBJECT_KEY);
+		assertEquals(mockBlob, result);
+		verify(mockStorage).get(OBJECT_BLOB_ID);
+	}
+
+	@Test
+	public void putObject() {
+		byte[] fileContents = "some file content".getBytes(StandardCharsets.UTF_8);
+		InputStream inputStream = new ByteArrayInputStream(fileContents);
+		// We use a stubbed class to verify that the contents are copied properly
+		WriteChannelStub writableByteChannel = new WriteChannelStub();
+		when(mockStorage.writer(OBJECT_BLOB_INFO)).thenReturn(writableByteChannel);
+		client.putObject(BUCKET_NAME, OBJECT_KEY, inputStream);
+		verify(mockStorage).writer(OBJECT_BLOB_INFO);
+		// Byte array equality checks the object, we can just convert back to a string.
+		assertArrayEquals(fileContents, writableByteChannel.outputStream.toByteArray());
+	}
+
+	@Test
+	public void deleteObjectSuccess() {
+		when(mockStorage.delete(OBJECT_BLOB_ID)).thenReturn(true);
+		client.deleteObject(BUCKET_NAME, OBJECT_KEY);
+		verify(mockStorage).delete(OBJECT_BLOB_ID);
+	}
+
+	@Test
+	public void deleteObjectFailure() {
+		when(mockStorage.delete(OBJECT_BLOB_ID)).thenReturn(false);
+		assertThrows(RuntimeException.class, () -> client.deleteObject(BUCKET_NAME, OBJECT_KEY));
```

Comment:
change runtime to another more specific exception type. we usually only use runtime exceptions to wrap checked exceptions that we don't expect to occur